otaclient/app/boot_control/_rpi_boot.py

```diff
@@ -0,0 +1,521 @@
+        self.system_boot_path = Path(cfg.SYSTEM_BOOT_MOUNT_POINT)
+        if not (
+            self.system_boot_path.is_dir()
+            and CMDHelperFuncs.is_target_mounted(self.system_boot_path)
+        ):
+            _err_msg = "system-boot is not presented or not mounted!"
+            logger.error(_err_msg)
+            raise BootControlInitError(_err_msg)
+        self._init_slots_info()
+        self._init_boot_files()
+
+    def _init_slots_info(self):
+        """Get current/standby slots info."""
+        logger.debug("checking and initializing slots info...")
+        try:
+            # detect active slot
+            self._active_slot_dev = CMDHelperFuncs.get_dev_by_mount_point(
+                cfg.ACTIVE_ROOTFS_PATH
+            )
+            if not (
+                _active_slot := CMDHelperFuncs.get_fslabel_by_dev(self._active_slot_dev)
+            ):
+                raise ValueError(
+                    f"failed to get slot_id(fslabel) for active slot dev({self._active_slot_dev})"
+                )
+            self._active_slot = _active_slot
+
+            # detect standby slot
+            # NOTE: using the similar logic like grub, detect the silibing dev
+            #       of the active slot as standby slot
+            _parent = CMDHelperFuncs.get_parent_dev(self._active_slot_dev)
+            # list children device file from parent device
+            # exclude parent dev(always in the front)
+            # expected raw result from lsblk:
+            # NAME="/dev/sdx"
+            # NAME="/dev/sdx1" # system-boot
+            # NAME="/dev/sdx2" # slot_a
+            # NAME="/dev/sdx3" # slot_b
+            _raw_child_partitions = CMDHelperFuncs._lsblk(f"-Pp -o NAME {_parent}")
+            try:
+                # NOTE: exclude the first 2 lines(parent and system-boot)
+                _child_partitions = list(
+                    map(
+                        lambda _raw: _raw.split("=")[-1].strip('"'),
+                        _raw_child_partitions.splitlines()[2:],
+                    )
+                )
+                if (
+                    len(_child_partitions) != 2
+                    or self._active_slot_dev not in _child_partitions
+                ):
+                    raise ValueError
+                _child_partitions.remove(self._active_slot_dev)
+            except Exception:
+                raise ValueError(
+                    f"unexpected partition layout: {_raw_child_partitions}"
+                )
+            # it is OK if standby_slot dev doesn't have fslabel or fslabel != standby_slot_id
+            # we will always set the fslabel
+            self._standby_slot = self.AB_FLIPS[self._active_slot]
+            self._standby_slot_dev = _child_partitions[0]
+            logger.info(
+                f"rpi_boot: active_slot: {self.active_slot}({self.active_slot_dev}), "
+                f"standby_slot: {self.standby_slot}({self.standby_slot_dev})"
```

Comment:
```suggestion
                f"rpi_boot: active_slot: {self._active_slot}({self._active_slot_dev}), "
                f"standby_slot: {self._standby_slot}({self._standby_slot_dev})"
```

Reply:
@Bodong-Yang I'd like to approve if this typo is fixed.

Reply:
@obi-t4 Sorry I focus too much on otaproxy side, I will resolve the PR comments here now.

Reply:
Updated in [a4a07d3](https://github.com/tier4/ota-client/pull/182/commits/a4a07d39d1d7988ee9796edde8a45ea8a0c9ddea).